File: backend/servicios/usuario_service.py
from repositorio import usuario_repository

def listar():
    return usuario_repository.listar()

def crear(u):
    return usuario_repository.crear(u)

def actualizar(u):
    return usuario_repository.actualizar(u)

def eliminar(id):
    return usuario_repository.eliminar(id)

def verificar_login(correo: str, password: str):
    usuarios = usuario_repository.listar()
    for u in usuarios:
        if u.get("correo") == correo and u.get("password") == password:
            return u
    return None

File: Programa_Final_Frontend/src/app/services/usuario.service.ts
import { Injectable } from '@angular/core';
import { BaseHttpService } from './base-http.service';
import { Usuario } from '../models/usuario.model';

@Injectable({ providedIn: 'root' })
export class UsuarioService {
  constructor(private http: BaseHttpService) {}

  listar(): Promise<Usuario[]> {
    return this.http.get<Usuario[]>('/usuarios/');
  }

  crear(usuario: Usuario): Promise<Usuario> {
    return this.http.post<Usuario>('/usuarios/', usuario);
  }

  actualizar(usuario: Usuario): Promise<Usuario> {
    return this.http.put<Usuario>(`/usuarios/${usuario.id}`, usuario);
  }

  eliminar(id: number): Promise<void> {
    return this.http.delete<void>(`/usuarios/${id}`);
  }
}
